### PR TITLE
PE-947: April 8th Executive, Oracles

### PR DESCRIPTION
**Pull request by godsflaw**
# Description

April 8th Executive, Oracles

# Contribution Checklist

- [x] PR title starts with `(PE-<TICKET_NUMBER>)`
- [ ] Code approved
- [ ] Tests approved
- [ ] CI Tests pass

# Checklist

- [x] Every contract variable/method declared as public/external private/internal
- [x] Consider if this PR needs the `officeHours` modifier override
- [x] Verify expiration (`30 days` unless otherwise specified)
- [x] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [x] Validate all addresses used are in Goerli changelog or known
- [x] Notify any external teams affected by the spell so they have the opportunity to review
- [ ] Deploy spell to Goerli `ETH_GAS="XXX" ETH_GAS_PRICE="YYY" make deploy`
- [ ] Ensure contract is verified on `Goerli` etherscan
- [ ] Change test to use Goerli spell address and deploy timestamp
- [ ] Cast spell on Goerli `make spell="0x-deployed-spell-address" cast-spell`
- [ ] Run `make archive-spell` or `make date="YYYY-MM-DD" archive-spell` to make an archive directory and copy `Goerli-DssSpell.sol`, `Goerli-DssSpell.t.sol`, `Goerli-DssSpell.t.base.sol`, and `Goerli-DssSpellCollateralOnboarding.sol`
- [ ] `squash and merge` this PR

### CODE REVIEW

File: src/Goerli-DssSpell.sol [hunk 1]
```diff
@@ -23,27 +23,119 @@ import "dss-exec-lib/DssAction.sol";
...
+    function list() external view returns (bytes32[] memory);
+    function pip(bytes32 ilk) external view returns (address);
+    function class(bytes32 ilk) external view returns (uint256);
+}
 
 contract DssSpellAction is DssAction, DssSpellCollateralOnboardingAction {
     // Provides a descriptive tag for bot consumption
     string public constant override description = "Goerli Spell";
 
-    // Turn office hours off
```
Comment (iamchrissmith) [inline src/Goerli-DssSpell.sol:31]: nit: it looks like you lost this comment but kept the function

File: src/Goerli-DssSpell.t.sol [hunk 2]
```diff
@@ -25,6 +25,223 @@ interface DenyProxyLike {
...
+        assertEq(OsmAbstract(0xF953cdebbbf63607EeBc556438d86F2e1d47C8aA).wards(ORACLE_WALLET01), 0);
+        assertEq(OsmAbstract(0x6Fb18806ff87B45220C2DB0941709142f2395069).wards(ORACLE_WALLET01), 0);
+        // assertEq(OsmAbstract(0x57A00620Ba1f5f81F20565ce72df4Ad695B389d7).wards(ORACLE_WALLET01), 0);
+        assertEq(OsmAbstract(0xCB772363E2DEc06942edbc5E697F4A9114B5989c).wards(ORACLE_WALLET01), 0);
+        assertEq(OsmAbstract(0x75B4e743772D25a7998F4230cb016ddCF2c52629).wards(ORACLE_WALLET01), 0);
+        assertEq(OsmAbstract(0x5AD3A560BB125d00db8E94915232BA8f6166967C).wards(ORACLE_WALLET01), 0);
+        assertEq(OsmAbstract(0x001eDD66a5Cc9268159Cf24F3dC0AdcE456AAAAb).wards(ORACLE_WALLET01), 0);
+        assertEq(OsmAbstract(0xDe112F61b823e776B3439f2F39AfF41f57993045).wards(ORACLE_WALLET01), 0);
+        // assertEq(OsmAbstract(0xdF8474337c9D3f66C0b71d31C7D3596E4F517457).wards(ORACLE_WALLET01), 0);
+        // assertEq(OsmAbstract(0xdF8474337c9D3f66C0b71d31C7D3596E4F517457).wards(ORACLE_WALLET01), 0);
```
Comment (iamchrissmith) [inline src/Goerli-DssSpell.t.sol:93]: why are these commented out?

File: src/Goerli-DssSpell.sol [hunk 3]
```diff
@@ -23,27 +23,119 @@ import "dss-exec-lib/DssAction.sol";
...
+    //
+
+    // --- Rates ---
+    uint256 constant ZERO_PCT_RATE           = 1000000000000000000000000000;
+    uint256 constant ZERO_ZERO_FIVE_PCT_RATE = 1000000000015850933588756013;
+    uint256 constant TWO_TWO_FIVE_PCT_RATE   = 1000000000705562181084137268;
+    uint256 constant THREE_TWO_FIVE_PCT_RATE = 1000000001014175731521720677;
+    uint256 constant FOUR_FIVE_PCT_RATE      = 1000000001395766281313196627;
+
+    address constant public ORACLE_WALLET01 = 0x4D6fbF888c374D7964D56144dE0C0cFBd49750D3;
```
Comment (brianmcmichael) [inline src/Goerli-DssSpell.sol:57]: I wonder if either of these should be added to `./src/test/addresses_deployers.sol`

File: src/Goerli-DssSpell.t.sol (hunk 2 above)
Comment (godsflaw) [reply src/Goerli-DssSpell.t.sol:93]: These represent the original ask from Oracles (see discord), but the original ask was wrong (as stables and RWA don't have `wards`).

File: src/Goerli-DssSpell.sol (hunk 3 above)
Comment (godsflaw) [reply src/Goerli-DssSpell.sol:57]: I took this approach first, but the deployers check ensures the deployers are no longer `wards`.  In goerli, this check is basically short circuited with a `continue`.

**Comment by iamchrissmith** [comment]
confirmed changes with the mainnet spreadsheet too

File: src/Goerli-DssSpell.sol (hunk 3 above)
Comment (brianmcmichael) [reply src/Goerli-DssSpell.sol:57]: That makes sense. We should probably add these addresses to `spell-mainnet`'s deployer list to ensure they aren't wards there.

File: src/Goerli-DssSpell.sol [hunk 4]
```diff
@@ -23,27 +23,120 @@ import "dss-exec-lib/DssAction.sol";
...
+        // ------------------------ OSM auth updates ---------------------------
+        // Adds two oracle wallets to all the OSMs
+        // https://discord.com/channels/893112320329396265/897479589171986434/960938185800683580
+        bytes32[] memory ilks = RegistryLike(DssExecLib.reg()).list();
+        for(uint256 i = 0; i < ilks.length; i++) {
+            uint256 class = RegistryLike(DssExecLib.reg()).class(ilks[i]);
+            if (class != 1) { continue; }
+
+            address pip = RegistryLike(DssExecLib.reg()).pip(ilks[i]);
+            // skip USDC, TUSD, PAXUSD, GUSD
```
Comment (brianmcmichael) [inline src/Goerli-DssSpell.sol:129]: What's the reason for skipping these?

If we're already giving them auth status on the OSM's what's the (additional) harm they can do by having access to these?

**Comment by godsflaw** [comment]
Deployed and fixed `make deploy-stamp`

```haskell
dapp-test: rpc block: latest
Running 15 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testFail_notScheduled() (gas: 12117)
[PASS] test_auth_in_sources() (gas: 6106509)
[PASS] test_use_eta() (gas: 338852)
[PASS] testOnTime() (gas: 2868942)
[PASS] test_nextCastTime() (gas: 339909)
[PASS] testFailTooEarly() (gas: 8618)
[PASS] test_oracle_list() (gas: 2978632)
[PASS] test_bytecode_matches() (gas: 1618074)
[PASS] test_auth() (gas: 12290410)
[PASS] testSpellIsCast_GENERAL() (gas: 12039727)
[PASS] testFailWrongDay() (gas: 8640)
[PASS] testCastCost() (gas: 2870509)
[PASS] testNewChainlogValues() (gas: 2878986)
[PASS] testFailTooLate() (gas: 8617)
[PASS] test_OSM_auth() (gas: 3256316)
```

**Comment by brianmcmichael** [comment]
Tests passing with deployed spell.

**Review by brianmcmichael — APPROVED** [review]

**Review by iamchrissmith — APPROVED** [review]
archive looks good

**Pull request merged**